tests/features/common-tools/common-tools.js

```diff
@@ -0,0 +1,13 @@
+module.exports = {
+  locatorBuilder: function(strings, ...keys) {
+    return function(...values) {
+      let dict = values[values.length - 1] || {}
+      let result = [strings[0]]
+      keys.forEach(function(key, i) {
+        let value = Number.isInteger(key) ? values[key] : dict[key]
```

Comment:
If you're not re-assigning a variable, declare it with `const` instead of `let`.
```suggestion
      const dict = values[values.length - 1] || {}
      const result = [strings[0]]
      keys.forEach(function(key, i) {
        const value = Number.isInteger(key) ? values[key] : dict[key]
```
Performing mutating methods on it (like `.push(value)`) does not mean you need to declare the variable with `let`. Only assignment operators (such as `=`, `+=`, `-=`, `*=`, `/=`, `%=`, `**=`, `||=`, `??=`, `&&=`), and increment (`++`) and decrement (`--`) operators will fail on `const` variables.

Please apply this in the rest of this PR.